begin switching in kitchen-salt for running the test suite

gtmanfred opened the pull request:
### What does this PR do?
Work on replacing salt-jenkins-build with kitchen-salt.

This will be used so that we are running the same environment as users for our unit tests, and so that they can replicate it.

gtmanfred committed:
run tests with kitchen

gtmanfred committed:
add script for copying back artifacts

gtmanfred committed:
add opennebula to Gemfile

gtmanfred committed:
test opennebula

gtmanfred committed:
fix output columns

gtmanfred committed:
which vagrant should go to stderr

gtmanfred committed:
use transport if not set in state_file

gtmanfred committed:
add logging

gtmanfred committed:
try newest salttesting

gtmanfred commented:
Go Go Jenkins!

gtmanfred committed:
fix pylint

thatch45 commented:
I think this is great, although the presence of a Gemfile kind of freaks me out, can that file be commented to specify that this is local to running the test suite? Should/Could that file be in `tests/` ?

gtmanfred commented:
We cannot move the gemfile and still use bundler.

Bundler requires the Gemfile to be in the directory where you are going to run the command after the install.

I can add a comment that this is only used for running the test suite.

gtmanfred commented:
We could just not use bundler, and do a gem install on the file, and use that in the test suite, but bundler seems to be pretty well established as how you should run your stuff from what I have seen in the ruby community.

s0undt3ch commented on tests/copyartifacts.py:
```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
```
Perhaps umder`tests/support`?

thatch45 commented:
I am ok with using a gemfile, I just think that we should make sure it is very clear that the gemfile is only used for the test suite

gtmanfred committed:
add comment to Gemfile and move copyartifacts

moved copyartifacts.py to tests/support

Also added a comment to the Gemfile that this is only used to run the testsuite with kitchen-salt

gtmanfred committed:
fix pylint comments

gtmanfred commented:
looks like the tests pass, I am not sure what gpg is doing, because all my commits are verified.
